theseus/optimizer/linearization.py

```diff
@@ -51,6 +51,7 @@ def linearize(self):
             raise RuntimeError(
                 "Attempted to linearize an objective with an incomplete variable order."
             )
+        self.objective.update_vectorization()
```

Comment:
Is the overhead for doing this roughly the same as what we had in the experimental branch?

Reply:
I expect the overhead to be similar, but we need to test it to be sure. Not sure if @fantaosha did some additional optimizations that I forgot to include. 

This does work with the linear optimizers.

Reply:
Sounds good. We'll know more when we rerun profiling.

Any easy way to make it more general so it works at the optimizer level? Besides linear optimizers we'll likely continue to add other alternatives like GBP that Joe is working on right now.

Reply:
Our current requirements is to call `Objective.update_vectorization()` before any work that wants to use the vectorized cost functions. We may have a better idea if there is some better place than linearization to automatically do these for other optimization classes when we have one or more implemented.

Reply:
We also need to do it after every iteration in the internal loop, so that alone wouldn't be sufficient.

Reply:
Ah right, okay let's figure it out in future PRs.